src/options.c

```diff
@@ -369,30 +414,73 @@ JOHN_USAGE_REGEX \
 "                           tunable cost parameters, see doc/OPTIONS\n" \
 "--save-memory=LEVEL        enable memory saving, at LEVEL 1..3\n" \
 "--node=MIN[-MAX]/TOTAL     this node's number range out of TOTAL count\n" \
+"--subformat=FORMAT         pick a benchmark format for --format=crypt\n" \
+"--mkpc=N                   request a lower max. keys per crypt\n" \
+"--min-length=N             request a minimum candidate length in bytes\n" \
+"--max-length=N             request a maximum candidate length in bytes\n" \
+"--length=N                 shortcut for --min-len=N --max-len=N\n" \
+"--field-separator-char=C   use 'C' instead of the ':' in input and pot files\n" \
+"--fix-state-delay=N        performance tweak, see doc/OPTIONS\n" \
+"--config=FILE              use FILE instead of john.conf or john.ini\n" \
+"--log-stderr               log to screen instead of file\n"             \
+"--verbosity=N              change verbosity (1-%u or %u for debug, default %u)\n" \
+"--no-log                   disables creation and writing to john.log file\n"      \
+"--bare-always-valid=Y      treat bare hashes as valid (Y/N)\n" \
+"--progress-every=N         emit a status line every N seconds\n" \
+"--crack-status             emit a status line whenever a password is cracked\n" \
+"--keep-guessing            try finding plaintext collisions\n" \
+"--no-keep-guessing         turn off any default \"keep guessing\" for used format\n" \
+"--max-candidates=[-]N      gracefully exit after this many candidates tried.\n" \
+"                           (if negative, reset count on each crack)\n" \
+"--max-run-time=[-]N        gracefully exit after this many seconds (if negative,\n" \
+"                           reset timer on each crack)\n" \
+"--regen-lost-salts=N       brute force unknown salts (see doc/OPTIONS)\n" \
 JOHN_USAGE_FORK \
 "--pot=NAME                 pot file to use\n" \
-"--list=WHAT                list capabilities, see --list=help or doc/OPTIONS\n"
+"--list=WHAT                list capabilities, see --list=help or doc/OPTIONS\n" \
+"--input-encoding=NAME      input encoding (alias for --encoding)\n" \
+"--internal-codepage=NAME   codepage used in rules/masks (see doc/ENCODINGS)\n" \
+"--target-encoding=NAME     output encoding (used by format, see doc/ENCODINGS)\n" \
+"--tune=HOW                 tuning options (auto/report/N)\n" \
 
 #define JOHN_USAGE_FORMAT \
-"--format=NAME              force hash of type NAME. The supported formats can\n" \
-"                           be seen with --list=formats and --list=subformats\n\n"
+"--format=[NAME|CLASS]      force hash of type NAME. The supported formats can\n" \
+"                           be seen with --list=formats and --list=subformats. Valid\n" \
+"                           classes: dynamic, cpu, opencl, ztex, mask, omp, all,\n" \
+"                           enabled, disabled.\n"
 
 #if defined(HAVE_OPENCL)
```

Comment:
Not sure from my current view (kinda confusing wall of strings and macros) but I think line 444 shouldn't end with a `\` although in this incarnation the blank line hides that problem.

Reply:
Great catch! You're indeed right that it's quite confusing to me. But I hope to acquaint myself with the codebase by working on issues like this one. :-)

I just pushed a fix. There's a few noisy commits now, so maybe I should squash the commits. I can't find any option here in GitHub for the pull request to squash before merge. Maybe that's on the reviewer/merger's side?